operator/internal/controller/auto/workspace_controller.go

```diff
@@ -434,6 +441,61 @@ func newStatefulSet(ctx context.Context, w *autov1alpha1.Workspace, source *sour
 /share/agent init -t /share/source --git-url $GIT_URL --git-revision $GIT_REVISION &&
 ln -s /share/source/$GIT_DIR /share/workspace
 		`
+		env := []corev1.EnvVar{
+			{
+				Name:  "GIT_URL",
+				Value: source.Git.URL,
+			},
+			{
+				Name:  "GIT_REVISION",
+				Value: source.Git.Ref,
+			},
+			{
+				Name:  "GIT_DIR",
+				Value: source.Git.Dir,
+			},
+		}
+		if source.Git.Shallow {
+			env = append(env, corev1.EnvVar{
+				Name:  "GIT_SHALLOW",
+				Value: "true",
+			})
+		}
+		// TODO: These are only available to the init container -- Should they
```

Comment:
I think no with the rationale that the vars should remain a private contract between the operator and the agent.

Reply:
Agree!